hazelcast/src/main/java/com/hazelcast/projection/Projection.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.projection;
+
+import java.io.Serializable;
+
+/**
+ * Enables transforming object into other objects.
+ * Examplay usage scenario is the project() method of the {@link com.hazelcast.core.IMap}
+ * <p>
+ * Only 1:1 transformations allowed. Use an Aggregator to perform N:1 aggregations.
+ *
+ * @param <I> input type
+ * @param <O> output type
+ */
+public abstract class Projection<I, O> implements Serializable {
```

Comment:
is it `Serializable` for easy-to-use in Java 8? (=projection via serializable lambda)
Otherwise I don't see a good reason to enforce `j.u.Serializable`